### ISSUE TITLE
`lu_unpack` crash with free(): invalid pointer

### ISSUE THREAD

**Issue by creakseek** [pytorch/pytorch#87962]
### 🐛 Describe the bug

`lu_unpack` abort with free(): invalid pointer
```
import torch
LU_data = torch.tensor([[1, 2, 3], [4, 5, 6], [7, 8, 9]], dtype=torch.float32)
LU_pivots = torch.tensor([0, 1, 2], dtype=torch.int32)
torch.lu_unpack(LU_data, LU_pivots, unpack_data=True, unpack_pivots=True)
```
```
free(): invalid pointer
abort (core dumped)
```

### Versions

pytorch 1.12.1

cc @jianyuh @nikitaved @pearu @mruberry @walterddr @IvanYashchuk @xwang233 @Lezcano

**Comment by malfet** [comment]
I think this is caused by the missing check that LU_pivots is essentially the empty tensors

**Comment by malfet** [comment]
Hmm, looks like this already been fixed in trunk/1.13:
```
 % python -c "import torch;torch.lu_unpack(torch.arange(9).view(3, 3)+1, torch.arange(3, dtype=torch.int32), unpack_pivots=True)"
Traceback (most recent call last):
  File "<string>", line 1, in <module>
RuntimeError: pivots passed to lu_unpack must be between 1 and LU.size(-2) inclusive.Did you properly pass the result of lu_factor?
```
